src/python/pants/core/goals/tailor.py

```diff
@@ -404,16 +413,48 @@ def make_content(bf_path: str, pts: Iterable[PutativeTarget]) -> FileContent:
     return EditedBuildFiles(new_digest, tuple(sorted(created)), tuple(sorted(updated)))
 
 
+def specs_to_dirs(specs: Specs) -> tuple[str, ...]:
+    """Extract cmd-line specs that look like directories.
+
+    Error on all other specs.
+
+    This is a hack that allows us to emulate "directory specs", until we are able to
+    support those more intrinsically.
+
+    TODO: If other goals need "directory specs", move this logic to a rule that produces them.
+    """
+    # Specs that look like directories are interpreted by the SpecsParser as the address
+    # of the target with the same name as the directory of the BUILD file.
+    # Note that we can't tell the difference between the user specifying foo/bar and the
+    # user specifying foo/bar:bar, so we will consider the latter a "directory spec" too.
+    if (
+        any(specs.filesystem_specs.includes)
+        or any(specs.filesystem_specs.ignores)
+        or any(specs.address_specs.globs)
+        or any(
+            spec.target_component != os.path.basename(spec.path_component)
+            for spec in specs.address_specs.literals
+        )
```

Comment:
You can create a better error message by using a list comprehension to find all offenders. If there are any, print the offenders in the error message:

> The tailor goal only accepts literal directories as arguments, but you specified [a, b c].
>
> Alternatively, use no arguments to run against the entire repository.

Reply:
Done.